debezium-storage/debezium-storage-kafka/src/main/java/io/debezium/storage/kafka/history/KafkaSchemaHistory.java

```diff
@@ -316,13 +322,16 @@ protected void recoverRecords(Consumer<HistoryRecord> records) {
                             "The database schema history couldn't be recovered. Consider to increase the value for " + RECOVERY_POLL_INTERVAL_MS.name());
                 }
 
+                checkForInterruption();
```

Comment:
The schema history recovery is moved out of task start and is now happening in the change-event-source-coordinator thread. If recovery time is high and task's stop method is called, it would send an interrupt to the change-event-source-coordinator thread(`Executor.shutdownNow`). It is important to catch this interrupt early and terminate the recovery process so that this thread is terminated. `checkForInterruption` has been put at certain places in the code where we are calling external code (KafkaConsumer). Reason being, if the thread is interrupted and is not handled properly (we can't be sure about interrupt handling in external code), the interrupt could be missed/cleared and thread would keep on running forever.